### PR TITLE
Reduce Azure Linux 3.0 images size

### PR DESCRIPTION
**Pull request by chalamalasetty**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Reduce Azure Linux 3.0 images size.
Due to perl and its dependency submodules Core images size bloated to 781 MB.
Perl was added as a runtime dependency for openssl in 3.0 which seems to be not needed.
Perl will be installed when openssl-perl package is installed.
Clean up tdnf cache from post install script in images build. 

###### Change Log  

- Change
   Perl will be installed with openssl-perl sub-package and not needed as default runtime dependency.

###### Does this affect the toolchain?  

**YES**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: Local build

### CODE REVIEW

File: toolkit/imageconfigs/postinstallscripts/core-container/cleanup.sh [hunk 1]
```diff
@@ -7,3 +7,7 @@ rm -rf /boot/*
 rm -rf /usr/src/
 rm -rf /home/*
 rm -rf /var/log/*
+
+echo removing tdnf cache
```
Comment (mbykhovtsev-ms) [inline toolkit/imageconfigs/postinstallscripts/core-container/cleanup.sh:11]: There is already a script that does it https://github.com/microsoft/azurelinux/blob/3.0-dev/toolkit/imageconfigs/postinstallscripts/remove-tdnf-cache.sh

But might I suggest imbedding this action at the end of build in our tooling instead? I have a branch that moves the script into our Go tooling itself, I just never got around to testing it. But after some discussion a while ago we determined its better for it to live in our build tooling and not post install scripts

Comment (chalamalasetty) [reply toolkit/imageconfigs/postinstallscripts/core-container/cleanup.sh:11]: Remove-tdnf-cache script is used for VHD/VHDX in images cleanup and this cleanup script is specific to core-conatiner.  I think yes it is good to add  tdnf cache clean-up from toolkit.

File: SPECS/openssl/openssl.spec [hunk 2]
```diff
@@ -9,7 +9,7 @@
 Summary: Utilities from the general purpose cryptography library with TLS implementation
 Name: openssl
 Version: 3.1.4
-Release: 1%{?dist}
+Release: 2%{?dist}
```
[the remark this answers is not in the document]
Comment (anphel31) [reply SPECS/openssl/openssl.spec:12]: Since this is a functional change to the package I think it's fine to bump the release

**Review by anphel31 — APPROVED** [review]

File: SPECS/openssl/openssl.spec (hunk 2 above)
[the remark this answers is not in the document]
Comment (tobiasb-ms) [reply SPECS/openssl/openssl.spec:12]: Not really a functional change -- removing a requires that it doesn't need (which, yes, decreases the overall size by a lot). And of course it's fine to bump it, but it also doesn't gain us anything.

**Pull request merged**